package/base-files/files/lib/functions/ipv4.sh

```diff
@@ -0,0 +1,142 @@
+#
+# rewrite of /bin/ipcalc.sh from awk to shell for use as a library
+# in init.d wrappers, and better portability (since busybox awk
+# lacks -E support at this time).
+#
+
+decimal=0
+
+bitcount() {
+    local c="$1"
+
+    c=$((((c >> 1) & 0x55555555) + (c & 0x55555555)))
+    c=$((((c >> 2) & 0x33333333) + (c & 0x33333333)))
+    c=$((((c >> 4) & 0x0f0f0f0f) + (c & 0x0f0f0f0f)))
+    c=$((((c >> 8) & 0x00ff00ff) + (c & 0x00ff00ff)))
+    c=$((((c >> 16) & 0x0000ffff) + (c & 0x0000ffff)))
+
+    echo "$c"
+}
+
+int32() {
+    local n="$1"
+
+    if [ -n "${n//[0-9]/}" ]; then
```

Comment:
This is new to me.
Afaik, the canonical way to to check for an integer with POSIX shell is `[ "$n" -eq "$n" ]`. It will be an easier read for those shell aficionados.

Reply:
It's actually used in a lot of places.  It's idiomatic.  You could use `[0-9a-f]` to check for hex digits, `[0-7]` for octal, etc.  Your suggestion only works with base 10.

Reply:
Alright! Interesting to know :-)

For POSIX compat, I would ask for `[:digit:]` - but if busybox has it, whatever.